controllers/cluster_controller.go

```diff
@@ -153,23 +153,37 @@ func (r *ClusterReconciler) SetupWithManager(mgr ctrl.Manager, log logr.Logger)
+// +kubebuilder:rbac:groups=anywhere.eks.amazonaws.com,resources=fluxconfigs,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=addons.cluster.x-k8s.io,resources=clusterresourcesets,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=addons.cluster.x-k8s.io,resources=clusterresourcesets/status,verbs=get;list;watch;create;update;patch
+// +kubebuilder:rbac:groups=apps,resources=daemonsets;daemonsets/status;deployments,verbs=get;list;watch
+// +kubebuilder:rbac:groups=bmc.tinkerbell.org,resources=machines;machines/status,verbs=get;list;watch
+// +kubebuilder:rbac:groups=bootstrap.cluster.x-k8s.io,resources=kubeadmconfigtemplates;kubeadmconfigtemplates/status,verbs=create;get;list;patch;update;watch
 // +kubebuilder:rbac:groups=cluster.x-k8s.io,resources=*,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=test,resources=test,verbs=get;list;watch;create;update;patch;delete;kill
+// +kubebuilder:rbac:groups=clusterctl.cluster.x-k8s.io,resources=providers,verbs=get;list;watch
+// +kubebuilder:rbac:groups=controlplane.cluster.x-k8s.io,resources=*,verbs=create;get;list;patch;update;watch
+// +kubebuilder:rbac:groups=coordination.k8s.io,resources=leases,verbs=create;get;list;update;watch;delete
 // +kubebuilder:rbac:groups=distro.eks.amazonaws.com,resources=releases,verbs=get;list;watch
+// +kubebuilder:rbac:groups=etcdcluster.cluster.x-k8s.io,resources=*,verbs=create;get;list;patch;update;watch
 // +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=awssnowclusters;awssnowmachinetemplates;awssnowippools;vsphereclusters;vspheremachinetemplates;dockerclusters;dockermachinetemplates;tinkerbellclusters;tinkerbellmachinetemplates;cloudstackclusters;cloudstackmachinetemplates;nutanixclusters;nutanixmachinetemplates,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups="",namespace=eksa-system,resources=secrets,verbs=delete;
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=cloudstackclusters/status;cloudstackmachinetemplates/status;dockerclusters/status;dockermachinetemplates/status;nutanixclusters/status;nutanixmachinetemplates/status;vsphereclusters/status;vspheremachinetemplates/status,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=packages.eks.amazonaws.com,resources=packages,verbs=create;delete;get;list;patch;update;watch;
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles;clusterroles/status;clusterrolebindings;clusterrolebindings/status;rolebindings;rolebindings/status,verbs=get;list;watch
```

Comment:
This one as well. How does adding these reduce controller permissions? I'm assuming that the default state is not a free for all.

Reply:
patch/update/create is removed from this resource

Reply:
Okay, I see. I didn't realize these were existing permissions on the manager role patch